Assets/MixedRealityToolkit/_Core/Definitions/Devices/InteractionMapping.cs

```diff
@@ -84,164 +84,30 @@ private set
-        }
-
-        #endregion Get Operators
+        #region Operators
 
-        #region Set Operators
-
-        public void SetValue(object newValue)
-        {
-            if (AxisType != AxisType.Raw)
-            {
-                Debug.LogError("SetValue(object) is only valid for AxisType.Raw InteractionMappings");
-            }
-
-            Changed = rawData != newValue;
-            rawData = newValue;
-        }
-
-        public void SetValue(bool newValue)
-        {
-            if (AxisType != AxisType.Digital)
-            {
-                Debug.LogError("SetValue(bool) is only valid for AxisType.Digital InteractionMappings");
-            }
-
-            Changed = boolData != newValue;
-            boolData = newValue;
-        }
-
-        public void SetValue(float newValue)
-        {
-            if (AxisType != AxisType.SingleAxis)
-            {
-                Debug.LogError("SetValue(float) is only valid for AxisType.SingleAxis InteractionMappings");
-            }
-
-            Changed = !floatData.Equals(newValue);
-            floatData = newValue;
-        }
-
-        public void SetValue(Vector2 newValue)
-        {
-            if (AxisType != AxisType.DualAxis)
-            {
-                Debug.LogError("SetValue(Vector2) is only valid for AxisType.DualAxis InteractionMappings");
-            }
-
-            Changed = vector2Data != newValue;
-            vector2Data = newValue;
-        }
-
-        public void SetValue(Vector3 newValue)
-        {
-            if (AxisType != AxisType.ThreeDoFPosition)
-            {
-                {
-                    Debug.LogError("SetValue(Vector3) is only valid for AxisType.ThreeDoFPosition InteractionMappings");
-                }
-            }
-
-            Changed = positionData != newValue;
-            positionData = newValue;
-        }
-
-        public void SetValue(Quaternion newValue)
-        {
-            if (AxisType != AxisType.ThreeDoFRotation)
-            {
-                Debug.LogError("SetValue(Quaternion) is only valid for AxisType.ThreeDoFRotation InteractionMappings");
-            }
-
-            Changed = rotationData != newValue;
-            rotationData = newValue;
-        }
+        /// <summary>
+        /// Set the value of the Interaction Mapping data
+        /// </summary>
+        /// <returns></returns>
+        public TReadingType GetValue() => currentReading;
 
-        public void SetValue(Tuple<Vector3, Quaternion> newValue)
+        /// <summary>
+        /// Get the value of the Interaction Mapping data
+        /// </summary>
+        /// <param name="newValue"></param>
+        public void GetValue(TReadingType newValue)
         {
-            if (AxisType != AxisType.SixDoF)
-            {
-                Debug.LogError("SetValue(Tuple<Vector3, Quaternion>) is only valid for AxisType.SixDoF InteractionMappings");
-            }
+            Changed = currentReading == null && newValue != null ||
```

Comment:
We could probably make it just currentReading != newValue.
Especially since value types can not be null.

Reply:
Tuple<T,T> is NOT a value type!, so we have to have this for this use case

Reply:
Changed = currentReading == null && newValue != null || 
           currentReading != null && newValue == null || 
           currentReading != null && newValue != null && 
           !currentReading.Equals(newValue);

It just feels like there is some redundancy here that isn't needed... I'll try a few things out in a simple app and update my comments.

Reply:
so, in the case of using just ==, Tuples with the same contents are considered different (ex: it is reference equals only).

in the case of value types, I need to confirm whether or not using a T to declare the type will report the same warnings as checking an explicitly defined one (ex: bool) for null.

Reply:
I see where using .Equals is required now (instances of T cannot use ==). That said, I still think we can make this check easier to read.

Reply:
I did some testing (please forgive the comment spam and subsequent deletions)… The following feels easier to read and should accomplish the same thing:

Changed = (currentReading != null) ? !current.Equals(newValue) : (newValue != null);

My results showed the following:
b1 (false) != b2 (true) True
b1 (false) != b3 (false) False

t1 (43, 7.15) != t2 (17, 4.3) True
t1 (43, 7.15) != t3 (43, 7.15) False
tn1 (null) != tn2 (null) False
tn1 (null) != t2 (17, 4.3) True
t1 (43, 7.15) != tn2 (null) True

Which matched the output from the version in the PR

Reply:
This is a ton of boxing in this op:
![image](https://user-images.githubusercontent.com/13334553/41496275-6f734122-70f0-11e8-9e78-c14fef7ee421.png)

Reply:
So, do you have a working alternative?
Noting we have to handle BOTH value and reference types

Reply:
Using direct methods 😝